pkg/zonemgr/zone_manager.go

```diff
@@ -28,52 +26,32 @@ func NewZoneManager() *ZoneManager {
 }
 
 func (zoneMgr *ZoneManager) init() {
-
-	fmt.Println("================ inside ZoneManager.init ===============")
-
 	//todo fetch custom details
-	clusterName := "cluster"
-	zoneFileName := "/zones/db.secdns"
-
-	userAdminEmail := ""
-	userSubdomain := "domain.com"
-	userNameserverName := ""
-	userNameserverIP := ""
-
-	zoneMgr.zoneFileCopy = NewZoneFile(clusterName, userAdminEmail,
-		userNameserverName, userNameserverIP, userSubdomain)
-
-	zoneMgr.zoneFileCopy.init()
+	adminEmail := ""
+	domain := ""
+	nameServerName := ""
+	nameServerIP := ""
 
-	//fmt.Println("================ zoneMgr.zoneFileCopy.content ===============\n", zoneMgr.zoneFileCopy.content)
+	zoneMgr.zoneFileCache = NewZoneFile(adminEmail,
+		nameServerName, nameServerIP, domain)
+	zoneMgr.zoneFileCache.init()
 
-	zoneMgr.zoneFileDns = NewDnsZoneFile(zoneFileName)
-	// todo return zoneFileDns.writeFile(zoneFileCopy.content) // is it necessary to create an empty zone file on load??
+	zoneMgr.zoneFileDns = NewDnsZoneFile(zoneFileName + zoneMgr.zoneFileCache.domain)
 }
 
 func (zoneMgr *ZoneManager) UpdateZone(namespacedName k8stypes.NamespacedName, interfaces []v1.VirtualMachineInstanceNetworkInterface) (err error) {
 
-	//fmt.Println("================ inside ZoneManager.UpdateZone ===============")
-	//fmt.Println("================ namespacedName: ===============", namespacedName)
-	//fmt.Println("================ interfaces: ===============", interfaces)
-
-	isUpdated, err := zoneMgr.zoneFileCopy.updateVMIRecords(namespacedName.Name, namespacedName.Namespace, interfaces)
-	if err != nil {
-		return err
-	}
-
-	//fmt.Println("================ zoneMgr.zoneFileCopy.content: ===============\n", zoneMgr.zoneFileCopy.content)
-	//fmt.Println("================ zoneMgr.zoneFileDns: ===============\n", zoneMgr.zoneFileDns)
+	isUpdated := zoneMgr.zoneFileCache.updateVMIRecords(namespacedName.Name, namespacedName.Namespace, interfaces)
 
 	//override on disk, create if not exist
 	if isUpdated {
-		err = zoneMgr.zoneFileDns.writeFile(zoneMgr.zoneFileCopy.content)
-		fmt.Println("********* err: ", err)
-
-		/////test
-		text, err := os.ReadFile("/zones/db.secdns")
-		fmt.Println("********* err: ", err)
-		fmt.Println("********* text: ", string(text))
+		err = zoneMgr.zoneFileDns.writeFile(zoneMgr.zoneFileCache.content)
+		/////test print - will be removed later
```

Comment:
Please remove those comments.
You can print to the log instead.

Reply:
done